pkg/action/archive.go

```diff
@@ -140,6 +140,26 @@ func extractTar(ctx context.Context, d string, f string) error {
 		return nil
 	case strings.Contains(filename, ".bz2") || strings.Contains(filename, ".bzip2"):
 		br := bzip2.NewReader(tf)
+		// Extract non-tar archives directly
+		if !strings.HasSuffix(filename, ".tar.bz2") {
+			uncompressed := strings.TrimSuffix(filepath.Base(f), ".bz2")
+			uncompressed = strings.TrimSuffix(uncompressed, ".bzip2")
+			target := filepath.Join(d, uncompressed)
+			if err := os.MkdirAll(filepath.Dir(target), 0o755); err != nil {
```

Comment:
Is world readable necessary? If not, maybe just `0700`?

Reply:
instead of `filepath.Dir(target)`, there is already `d`; is it sufficient?

Reply:
Updated in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).